### PR TITLE
Core - implemented transaction syntactic checks

### LINKED ISSUE #345: Core - syntactic validation methods
## Description

implement syntactic methods for validation and diagnostic 
* `bool output_basic_syntactic(output_basic_t* o)`, [Basic](https://github.com/lzpap/tips/blob/master/tips/TIP-0018/tip-0018.md#additional-transaction-syntactic-validation-rules)
* `bool output_alias_syntactic(output_alias_t* o)`, [Alias](https://github.com/lzpap/tips/blob/master/tips/TIP-0018/tip-0018.md#output-syntactic-validation)
* `bool output_foundry_syntactic(output_foundry_t* o)`, [Foundry](https://github.com/lzpap/tips/blob/master/tips/TIP-0018/tip-0018.md#output-syntactic-validation-1)
* `bool output_nft_syntactic(output_nft_t* o)`, [NFT](https://github.com/lzpap/tips/blob/master/tips/TIP-0018/tip-0018.md#output-syntactic-validation-2)
* `bool tx_payload_syntactic(transaction_payload_t* tx)`, [Transaction](https://github.com/lzpap/tips/blob/tx-updates/tips/TIP-0020/tip-0020.md#syntactic-validation)

### PR DESCRIPTION
**Pull request by oopsmonk**
# Description of change

* fixes #345 
* implemented Immutable Alias Address Unlock Condition 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

local unit tests passed, test_transaction.c must be updated to verify `tx_payload_syntactic()`

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/core/models/outputs/outputs.c [hunk 1]
```diff
@@ -349,3 +350,43 @@ void utxo_outputs_print(utxo_outputs_list_t *outputs, uint8_t indentation) {
...
+        printf("[%s:%d] deprecated or unsupported output type\n", __func__, __LINE__);
+        is_valid = false;
+        break;
+    }
+
+    if (is_valid == false) {
+      return is_valid;
+    } else {
+      // amount must fulfill the storage protection and must not be zero
+      return storage_deposit_check(byte_cost, elm->output->output_type, elm->output->output);
```
Comment (matjazv) [inline src/core/models/outputs/outputs.c]: After first element function already returns and other elements are not checked.

**Review by matjazv — CHANGES_REQUESTED** [review]

**Comment by matjazv** [comment]
Is somewhere in the code check for maximum number of outputs (`UTXO_OUTPUT_MAX_COUNT`)?

File: src/core/models/outputs/outputs.c (hunk 1 above)
Comment (oopsmonk) [reply src/core/models/outputs/outputs.c]: nice, thanks.

**Comment by oopsmonk** [comment]
> Is somewhere in the code check for maximum number of outputs (`UTXO_OUTPUT_MAX_COUNT`)?

it's checked when adding an element to the list in `utxo_outputs_add`, same as others, say native tokens, inputs

**Review by matjazv — APPROVED** [review]

**Review by rebinmannady93 — APPROVED** [review]

**Pull request merged**